linkedlist/src/main/java/ua/hillel/LinkedList.java

```diff
@@ -1,69 +1,163 @@
+import java.util.Objects;
+
 /**
  * Implement doubly linked list that is based on {@link Node<T>} class (you should implement it as well).
  * This is a simplified version of {@link java.util.LinkedList}.
  */
 public class LinkedList<T> implements List<T> {
+    private static class Node<T> {
+        T element;
+        Node<T> prev;
+        Node<T> next;
+
+        public Node(T element) {
+            this.element = element;
+            this.next = null;
+            this.prev = null;
+        }
+    }
 
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private Node<T> getNode(int index) {
+        Node<T> currNode = head;
+        for (int i = 0; i < index; i++) {
+            currNode = currNode.next;
+        }
+        return currNode;
+    }
 
     @Override
     public void add(T element) {
-
+        Node<T> newElement = new Node<>(element);
+        if (head == null) {
+            head = tail = newElement;
+        } else {
+            tail.next = newElement;
+            tail = newElement;
+        }
+        size++;
     }
 
     @Override
     public void add(int index, T element) {
-
+        Objects.checkIndex(index, size + 1);
+        Node<T> newElement = new Node<>(element);
+        if (head == null || index == size) {
+            add(element);
+        } else if (index == 0) {
+            newElement.next = head;
+            head = newElement;
+        } else {
+            Node<T> prevIndex = getNode(index - 1);
+            newElement.next = prevIndex.next;
+            prevIndex.next = newElement;
+        }
+        size++;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Objects.checkIndex(index, size);
+        return getNode(index).element;
     }
 
     @Override
     public T get(T element) {
+        Node<T> currElement = head;
+        while (currElement != null) {
+            if (currElement.element.equals(element)) {
+                return element;
+            }
+            currElement = currElement.next;
+        }
         return null;
     }
 
     @Override
     public T getFirst() {
-        return null;
+        if (isEmpty()) {
+            throw new NoSuchElementException();
+        }
+        return head.element;
     }
 
     @Override
     public T getLast() {
-        return null;
+        if (isEmpty()) {
+            throw new NoSuchElementException();
+        }
+        return tail.element;
     }
 
     @Override
     public void set(int index, T element) {
-
+        Objects.checkIndex(index, size);
+        Node<T> node = getNode(index);
+        node.element = element;
     }
 
     @Override
     public boolean remove(int index) {
-        return false;
+        if (index < 0 || index > size) {
+            return false;
+        }
+        Node<T> currElement = getNode(index);
+        if (size == 1) {
+            head = null;
+            tail = null;
+        } else if (currElement == head) {
```

Comment:
It's not correct to compare nodes by `==`, so if you're going to compare 2 nodes, take an element from it and compare. But here you shouldn't do that. This method has an index as a parameter, why don't you compare `index == 0` to check if the first element is going to be deleted?